docs/use_cases.md

```diff
@@ -0,0 +1,74 @@
+# Node.js Use Cases
+
+This doc is intended to capture the key uses cases for Nodej.js. These
+use cases will then be used to guide what we develop/run as part of
+the community benchmarking effort.
+
+# Main Use Case Categories
+
+The folloing categories have been identifed:
+
+* REST or REST like back end services
+* Generating/serving dynamic web page content
+* AJAX like applications with bidirectional communication over
+  websockets and/or http2
+* APIs/microservices based applications
+* Mobile back ends
+* Small scripts
+
+For these use cases one or more of the following are often imporant:
+
+* consistent low latency
+* ability to support high concurrency
+* throughput
+* fast startup/restart/shutdown
+* low resource usage (memory/cpu)
+
+## REST or REST like back end services
+
+Applications using REST
+(https://en.wikipedia.org/wiki/Representational_state_transfer)
+provide access to services over HTTP with verbs such as GET, POST,
+DELETE where URIs are used to identify specific resources.
+Key elements are accepting requests for a given URI and serving
+responses, often using JSON to exchange data.
+
+## Generating/serving dynamic web page content
+
+These applications are often built with additional modules like
+[express](http://expressjs.com), [hapi](http://hapijs.com)
+and [React](http://facebook.github.io/react/).  Key elements are
+building and returning dynamic web pages often using template engines.
+
+## AJAX like applications withh bidirectional communication over websockets and/or http2
+
+As an evolution of [AJAX](https://en.wikipedia.org/wiki/Ajax_\(programming\))
+type approaches, Node.js make it easy to build applications
+where the GUI for an application is delivered through an initial http
+request and then subsequent updates are managed through bidirectional
+messages between the client and the server using websockets or http2.
+Key elements are the speed and latency with which messages
+can be sent back and forth between client and server.
+
+## APIs/microservices based applications
+
+These applications consist of a number of APIs/microservices that are 
+co-ordinated to provide the resulting end user application.  Key elements are
+how well multiple small services can be run concurrently(low memory/cpu)
+and the speed/latency with which requests can be exchanged between the
+main applicaitons and APIs/microservices.
+
+## Mobile back ends
+
+These applications may provide elements of the other use cases, but the key 
+element is that the application can effectively support multiple mobile 
+platforms while achieving high concurrency and low latency. Also important is
+to minimize the size of the data that must be  communicated between
+the server and client.
+
+## Small scripts
```

Comment:
Worth expanding on this to discuss build tooling, particularly for front-end engineering, this is an absolutely massive area for Node.js that we generally overlook because the bulk of Node's users in this space don't even think of themselves as Node users.